janusgraph-hbase-parent/janusgraph-hbase-core/src/main/java/org/janusgraph/diskstorage/hbase/HBaseStoreManager.java

```diff
@@ -891,21 +896,45 @@ private void setCFOptions(HColumnDescriptor cdesc, int ttlInSeconds) {
                     }
 
                     for (StaticBuffer b : mutation.getDeletions()) {
+                        // commands.getSecond() is a Delete for this rowkey.
                         commands.getSecond().deleteColumns(cfName, b.as(StaticBuffer.ARRAY_FACTORY), delTimestamp);
                     }
                 }
 
                 if (mutation.hasAdditions()) {
-                    if (commands.getFirst() == null) {
-                        Put p = new Put(key, putTimestamp);
-                        commands.setFirst(p);
-                    }
-
+                    // All the entries (column cells) with the rowkey use this one Put, except the ones with TTL.
+                    Put p = new Put(key, putTimestamp);
+                    
                     for (Entry e : mutation.getAdditions()) {
-                        commands.getFirst().add(cfName,
+
+                        // Deal with TTL within the entry (column cell) first
+                        // HBase cell level TTL is actually set at the Mutation/Put level.
+                        // Therefore we need to construct a new Put for each entry (column cell) with TTL.
+                        // We can not combine them because column cells within the same rowkey may:
+                        // 1. have no TTL
+                        // 2. have TTL
+                        // 3. have different TTL
+                        Integer ttl = (Integer) e.getMetaData().get(EntryMetaData.TTL);
```

Comment:
Finally, dataType and isIdentifying return constant values for each of the three Enums. Can you create a package private constructor that just sets the values as part of the state of each enum instead of switch-casing?